### PR TITLE
Do not generate empty buckets for the date histogram

### LINKED ISSUE #88800: [CI] RollupActionSingleNodeTests testRollupDatastream failing
**Build scan:**
https://gradle-enterprise.elastic.co/s/xl4uqehihgkxs/tests/:x-pack:plugin:rollup:test/org.elasticsearch.xpack.rollup.v2.RollupActionSingleNodeTests/testRollupDatastream

**Reproduction line:**
`./gradlew ':x-pack:plugin:rollup:test' --tests "org.elasticsearch.xpack.rollup.v2.RollupActionSingleNodeTests.testRollupDatastream" -Dtests.seed=4E779F8B2A344C56 -Dtests.locale=sl -Dtests.timezone=NST -Druntime.java=18`

**Applicable branches:**
main

**Reproduces locally?:**
No

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.xpack.rollup.v2.RollupActionSingleNodeTests&tests.test=testRollupDatastream

**Failure excerpt:**
```
org.elasticsearch.action.search.SearchPhaseExecutionException: 
  at __randomizedtesting.SeedInfo.seed([4E779F8B2A344C56:84AC3906E3AE0A19]:0)
  at org.elasticsearch.action.search.AbstractSearchAsyncAction.onPhaseFailure(AbstractSearchAsyncAction.java:728)
  at org.elasticsearch.action.search.FetchSearchPhase$1.onFailure(FetchSearchPhase.java:95)
  at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:28)
  at org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:33)
  at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:769)
  at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
  at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
  at java.lang.Thread.run(Thread.java:833)

  Caused by: org.elasticsearch.search.aggregations.MultiBucketConsumerService$TooManyBucketsException: Trying to create too many buckets. Must be less than or equal to: [65536] but this number of buckets was exceeded. This limit can be set by changing the [search.max_buckets] cluster level setting.

    at org.elasticsearch.search.aggregations.MultiBucketConsumerService$MultiBucketConsumer.accept(MultiBucketConsumerService.java:115)
    at org.elasticsearch.search.aggregations.AggregationReduceContext$ForFinal.consumeBucketCountAndMaybeBreak(AggregationReduceContext.java:213)
    at org.elasticsearch.search.aggregations.AggregationReduceContext.consumeBucketsAndMaybeBreak(AggregationReduceContext.java:119)
    at org.elasticsearch.search.aggregations.bucket.histogram.InternalDateHistogram.addEmptyBuckets(InternalDateHistogram.java:407)
    at org.elasticsearch.search.aggregations.bucket.histogram.InternalDateHistogram.reduce(InternalDateHistogram.java:476)
    at org.elasticsearch.search.aggregations.InternalAggregations.reduce(InternalAggregations.java:159)
    at org.elasticsearch.search.aggregations.bucket.terms.AbstractInternalTerms.reduceBucket(AbstractInternalTerms.java:109)
    at org.elasticsearch.search.aggregations.DelayedBucket.reduced(DelayedBucket.java:59)
    at org.elasticsearch.search.aggregations.TopBucketBuilder$BufferingTopBucketBuilder.build(TopBucketBuilder.java:180)
    at org.elasticsearch.search.aggregations.bucket.terms.AbstractInternalTerms.reduce(AbstractInternalTerms.java:315)
    at org.elasticsearch.search.aggregations.InternalAggregations.reduce(InternalAggregations.java:159)
    at org.elasticsearch.search.aggregations.InternalAggregations.topLevelReduce(InternalAggregations.java:106)
    at org.elasticsearch.action.search.SearchPhaseController.reduceAggs(SearchPhaseController.java:601)
    at org.elasticsearch.action.search.SearchPhaseController.reducedQueryPhase(SearchPhaseController.java:570)
    at org.elasticsearch.action.search.QueryPhaseResultConsumer.reduce(QueryPhaseResultConsumer.java:130)
    at org.elasticsearch.action.search.FetchSearchPhase.innerRun(FetchSearchPhase.java:104)
    at org.elasticsearch.action.search.FetchSearchPhase$1.doRun(FetchSearchPhase.java:90)
    at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
    at org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:33)
    at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:769)
    at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
    at java.lang.Thread.run(Thread.java:833)

```

### PR DESCRIPTION
**Pull request by salvatore-campagna**
If the date histogram interval is large and the 'fixed_interval'
parameter is very small we might end up with a large number of
buckets in the resulting histogram, in case we also generate empty
buckets. Roughly (max date - min date) / fixed_interval > 65536.

Here we set minDocCount to 1 so to avoid generation of empty buckets.
In the test the maximum value for 'docCount' is 9000 which means,
in the worst case, we generate 9000 documents, each belonging to a
different bucket. In the worst case we would have 9000 buckets
maximum which is well below the maximum number of buckets
allowed by default (65536).

Resolves #88800.

### CODE REVIEW

**Comment by salvatore-campagna** [comment]
I tested this running the test locally "until failure". After more than 1000 executions I don't see any failure. Before the patch I could see the failure fairly quickly after a few tens executions (depending on random values).

**Comment by salvatore-campagna** [comment]
@elasticsearchmachine update branch

**Comment by salvatore-campagna** [comment]
@elasticsearchmachine update branch

File: docs/changelog/89070.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+pr: 89070
+summary: "Fix: do not generate empty buckets for the date histogram"
+area: "Rollup"
+type: bug
```
Comment (csoulios) [inline docs/changelog/89070.yaml:4]: This PR should be labeled as `>test-failure` (not a `bug`) and it should contain no changelog file

**Review by csoulios — APPROVED** [review]
LGTM!

I left a comment about labelling this PR. You do not need another review after fixing this

**Comment by salvatore-campagna** [comment]
@elasticsearchmachine update branch

**Comment by salvatore-campagna** [comment]
@elasticsearchmachine test this please

**Comment by salvatore-campagna** [comment]
@elasticsearchmachine update branch

**Comment by salvatore-campagna** [comment]
@elasticsearchmachine run elasticsearch-ci/part-2

**Comment by salvatore-campagna** [comment]
@elasticsearchmachine update branch

**Pull request merged**